### ISSUE TITLE
Enhance our schema hover link to jump to the right table

### ISSUE THREAD

**Issue by vinistock** [Shopify/ruby-lsp-rails#52]
Currently, our schema link just points to the file. However, we can link to the specific table for the model, which is a richer experience.

One possibility of how we can do this is by
1. Start returning the table name from our middleware
2. During the `activate` call parse `schema.rb` and create a hash of table names to locations
3. When executing hover, search the hash for the right line to jump to

**Concerns**

- If there are modifications in schema.rb, we would need to rebuild the hash. For that, we need [file watching](https://github.com/Shopify/ruby-lsp/pull/632) to be merged in the Ruby LSP and then we have to design a way in which extensions can ask the Ruby LSP to notify them when certain files are changed.
- This won't work for `structure.sql`. We probably don't want a SQL parser, but we might be able to achieve something with regexes

**Comment by faraazahmad** [comment]
Thanks! Although in this project I'm not storing the locations of the tables, I'm using `rubocop-ast` to parse the source. This gives you access to the path and location in the file for every node.

So in the schema.rb the approach would be to:

1. Look for `create_table` method calls (lets call it the `ddl_node`)
2. Constantize its first argument using ActiveSupport's string helpers and get the model name
3. Get children `line`, `column`, `last_line`, `last_column` from `ddl_node.location`

I have a [`Visitor::Schema`](https://github.com/faraazahmad/trains/blob/master/lib/trains/visitor/schema.rb) that I _borrowed_ from `rubocop-rails`'s [`SchemaLoader`](https://github.com/rubocop/rubocop-rails/blob/master/lib/rubocop/rails/schema_loader.rb) that basically does that, except for the location storing part.

I hope this makes sense.

**Comment by vinistock** [comment]
Yeah, that makes sense. Although we'd use Prism and not `rubocop-ast` here. So you can probably do something like

```ruby
class SchemaCollector < Prism::Visitor
  def initialize
    @tables = {}
  end

  def visit_call_node(node)
    # check if it's a `create_table` call
    # add it to the hash of tables
  end
end
```
